x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/transforms.ts

```diff
@@ -17,81 +23,84 @@ export const transformESSearchToKnowledgeBaseEntry = (
-      timestamp: kbEntrySchema['@timestamp'],
-      id: kbEntrySchema.id,
-      createdAt: kbEntrySchema.created_at,
-      createdBy: kbEntrySchema.created_by,
-      updatedAt: kbEntrySchema.updated_at,
-      updatedBy: kbEntrySchema.updated_by,
+    return transformEsSchemaToEntry(kbEntrySchema);
+  });
+};
+
+const transformEsSchemaToEntry = (
+  esKbEntry: EsKnowledgeBaseEntrySchema
+): DocumentEntry | IndexEntry => {
+  if (esKbEntry.type === DocumentEntryType.value) {
+    const documentEntry: DocumentEntry = {
+      id: esKbEntry.id,
+      createdAt: esKbEntry.created_at,
+      createdBy: esKbEntry.created_by,
+      updatedAt: esKbEntry.updated_at,
+      updatedBy: esKbEntry.updated_by,
       users:
-        kbEntrySchema.users?.map((user) => ({
+        esKbEntry.users?.map((user) => ({
           id: user.id,
           name: user.name,
         })) ?? [],
-      ...(kbEntrySchema.metadata
-        ? {
-            metadata: {
-              kbResource: kbEntrySchema.metadata.kbResource,
-              source: kbEntrySchema.metadata.source,
-              required: kbEntrySchema.metadata.required,
-            },
-          }
-        : {}),
-      namespace: kbEntrySchema.namespace,
-      text: kbEntrySchema.text,
-      ...(kbEntrySchema.vector
+
+      name: esKbEntry.name,
+      namespace: esKbEntry.namespace,
+      type: esKbEntry.type,
+      kbResource: esKbEntry.kb_resource,
+      source: esKbEntry.source,
+      required: esKbEntry.required,
+      text: esKbEntry.text,
+      ...(esKbEntry.vector
         ? {
             vector: {
-              modelId: kbEntrySchema.vector.model_id,
-              tokens: kbEntrySchema.vector.tokens,
+              modelId: esKbEntry.vector.model_id,
+              tokens: esKbEntry.vector.tokens,
             },
           }
         : {}),
     };
-
-    return kbEntry;
-  });
+    return documentEntry;
+  } else if (esKbEntry.type === IndexEntryType.value) {
+    const indexEntry: IndexEntry = {
+      id: esKbEntry.id,
+      createdAt: esKbEntry.created_at,
+      createdBy: esKbEntry.created_by,
+      updatedAt: esKbEntry.updated_at,
+      updatedBy: esKbEntry.updated_by,
+      users:
+        esKbEntry.users?.map((user) => ({
+          id: user.id,
+          name: user.name,
+        })) ?? [],
+      name: esKbEntry.name,
+      namespace: esKbEntry.namespace,
+      // Document Entry Fields
+      type: esKbEntry.type,
+      index: esKbEntry.index,
+      field: esKbEntry.field,
+      description: esKbEntry.description,
+      queryDescription: esKbEntry.query_description,
+      inputSchema:
+        esKbEntry.input_schema?.map((schema) => ({
+          fieldName: schema.field_name,
+          fieldType: schema.field_type,
+          description: schema.description,
+        })) ?? [],
+      outputFields: esKbEntry.output_fields ?? [],
+    };
+    return indexEntry;
+  }
+  throw new Error(`Unknown Knowledge Base Entry`);
```

Comment:
nit: No change requested, but when I followed the chain of function calls that use this function, i.e. `x-pack/plugins/elastic_assistant/server/routes/knowledge_base/entries/find_route.ts` it appears a single document with an unexpected `esKbEntry.type` may cause the retrieval of all other (matching) documents to fail.

Consider returning `null` and forcing the caller to deal with errors via `null` checks if it's useful for the caller to still get the other (valid) docs.

Reply:
Good call! I've added a `TODO` and will look at this when wiring up the UI and making the remainder of the API changes to support bulk operations.